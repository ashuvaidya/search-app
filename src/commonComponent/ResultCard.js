import propTypes from "prop-types";

export const ResultCard = ({ data, id }) => {
  const { title, description, tags, weight } = data;

  return (
    <div className="card-data">
      <div className="card-title" aria-label="title">
        <strong>{title}</strong>
      </div>
      <div className="card-content" aria-label="description">
        {description}
      </div>
      <div className="card-content" aria-label="tags">
        <span>Tags: </span>
        <ul>
          {tags?.map((tag, index) => {
            return <li key={`${id}-${index}`}>{tag}</li>;
          })}
        </ul>
      </div>
      <div className="card-content" aria-label="weight">
        <span>Weight: </span> <strong>{weight}</strong>
      </div>
    </div>
  );
};

ResultCard.propTypes = {
  title: propTypes.string,
  description: propTypes.string,
  tags: propTypes.array,
  weight: propTypes.number,
};
