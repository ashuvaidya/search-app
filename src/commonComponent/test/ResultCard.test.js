import React from "react";
import { render } from "@testing-library/react";
import { ResultCard } from "../ResultCard";

describe("testing for result card component", () => {
  it("render result card component", () => {
    const props = {
      data: {
        title: "test",
        description: "test",
        tags: ["test"],
        weight: 0,
      },
      id: "test",
    };
    const { container } = render(<ResultCard {...props} />);
    expect(container).toMatchSnapshot();
  });
});
