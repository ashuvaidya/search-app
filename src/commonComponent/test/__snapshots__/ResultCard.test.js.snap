// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing for result card component render result card component 1`] = `
<div>
  <div
    class="card-data"
  >
    <div
      aria-label="title"
      class="card-title"
    >
      <strong>
        test
      </strong>
    </div>
    <div
      aria-label="description"
      class="card-content"
    >
      test
    </div>
    <div
      aria-label="tags"
      class="card-content"
    >
      <span>
        Tags: 
      </span>
      <ul>
        <li>
          test
        </li>
      </ul>
    </div>
    <div
      aria-label="weight"
      class="card-content"
    >
      <span>
        Weight: 
      </span>
       
      <strong>
        0
      </strong>
    </div>
  </div>
</div>
`;
